### PR TITLE
improve units in modeling

### PR DESCRIPTION
**Pull request by nden**
@astrofrog This PR introduces `Model._param_metrics[param_name]['raw_unit']`. `raw_unit` is the unit corresponding to `Parameter._raw_value`. I felt this is needed because `Parameter.unit` is the unit of the user supplied input parameter, which may be different from the unit of `Parameter._raw_value` after a `Parameter._setter` executes.  This is important because the output of `Model._param_sets(raw=True, units=True)` is passed to `Model.evaluate` and needs to have the correct units, i.e. `raw_unit`.

With this cleared the rest of the changes involve corrections to `Parameter.value` which returns the user supplied value * unit (after accounting for `Parameter._getter`) and improvements to rotations with units. Also enabled units in `Shift` and `Scale`. 

One other question is the units of projections. I think the names `Pix2Sky` and `Sky2pix` were meant to indicate the direction of the transform. That doesn't mean the units are necessarily pixels. For example, the `TAN` projection is in angle units (not sure about the rest of the Zenithal).

TODO:
[x ] Fix units of projections in the "pixel" frame
[x ] Allow for angle units in general, do not assume rad or deg

### CODE REVIEW

**Comment by pllim** [comment]
If this gets in v2, it does not really need a change log.

**Comment by Cadair** [comment]
@nden The issue with the units on projections is that the units are not preserved through the evaluation of the model, so you have to have a predetermined output unit.

**Comment by nden** [comment]
@Cadair Agreed. What I am saying is that, at least in the case of a tangent projection, the input units of `Pix2Sky_TAN` and the output units of `Sky2Pix_TAN` should be angle units, not pixels, as a great circle on the sky projects to a straight line in the tangent plane. In addition we should allow any angle units as input, not just deg. I can push a fix for the tangent projection now. However, I'm not sure about the rest of the projections, need to look them up.

**Comment by nden** [comment]
The transforms in the `projections` module implement WCS Paper II. All functions in the paper are written in a way that requires inputs to be in deg and the return values are in deg. I changed `input_units` and `return_units` accordingly.

File: astropy/modeling/utils.py [hunk 1]
```diff
@@ -634,3 +634,19 @@ def _combine_equivalency_dict(keys, eq1=None, eq2=None):
...
+def _to_radian(value):
+    """ Convert ``value`` to radian. """
+    if isinstance(value, u.Quantity):
+        return value.to(u.rad)
+    else:
+        return np.deg2rad(value)
+
+
+def _to_orig_unit(value, raw_unit=None, orig_unit=None):
+    """ Conver value with ``raw_unit`` to ``orig_unit``. """
```
Comment (astrofrog) [inline astropy/modeling/utils.py]: Conver -> Convert

**Pull request merged**